glustercli/cmd/snapshot-list.go

```diff
@@ -36,13 +37,22 @@ func snapshotListHandler(cmd *cobra.Command) error {
 	table.SetAutoMergeCells(true)
 	table.SetRowLine(true)
 	if volname == "" {
+		if len(snaps) == 0 {
+			fmt.Println("There are no snapshots in the system")
+			return nil
+		}
 		table.SetHeader([]string{"Name", "Origin Volume"})
 		for _, snap := range snaps {
 			for _, entry := range snap.SnapName {
 				table.Append([]string{entry, snap.ParentName})
 			}
 		}
 	} else {
+		if len(snaps[0].SnapName) == 0 {
```

Comment:
i think we need to add `index` validation here

Reply:
Validation is not required here. Because if you have given volume name to list, then it is guaranteed to have one entry in the list. But that volume can have zero snapshots, hence the check.